src/actions/signage-actions.js

```diff
@@ -0,0 +1,425 @@
+/**
+ * Copyright 2022 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import {
+    getRequest,
+    createAction,
+    stopLoading,
+    startLoading,
+    authErrorHandler,
+    escapeFilterValue,
+    putRequest,
+    postRequest,
+    deleteRequest,
+    fetchErrorHandler, showSuccessMessage
+} from 'openstack-uicore-foundation/lib/utils/actions';
+import {getAccessTokenSafely} from '../utils/methods';
+import Ably from "ably";
+import T from "i18n-react";
+
+export const REQUEST_SIGN = 'REQUEST_SIGN';
+export const RECEIVE_SIGN = 'RECEIVE_SIGN';
+export const UPDATE_SIGN = 'UPDATE_SIGN';
+export const SIGN_UPDATED = 'SIGN_UPDATED';
+export const SIGN_ADDED = 'SIGN_ADDED';
+export const REQUEST_SIGNAGE_EVENTS = 'REQUEST_SIGNAGE_EVENTS';
+export const RECEIVE_SIGNAGE_EVENTS = 'RECEIVE_SIGNAGE_EVENTS';
+export const RECEIVE_SIGNAGE_TEMPLATES = 'RECEIVE_SIGNAGE_TEMPLATES';
+export const REQUEST_SIGNAGE_BANNERS = 'REQUEST_SIGNAGE_BANNERS';
+export const RECEIVE_SIGNAGE_BANNERS = 'RECEIVE_SIGNAGE_BANNERS';
+export const REQUEST_SIGNAGE_LOCATIONS = 'REQUEST_SIGNAGE_LOCATIONS';
+export const RECEIVE_SIGNAGE_LOCATIONS = 'RECEIVE_SIGNAGE_LOCATIONS';
+export const UPDATE_SIGNAGE_BANNER = 'UPDATE_SIGNAGE_BANNER';
+export const SIGNAGE_BANNER_UPDATED = 'SIGNAGE_BANNER_UPDATED';
+export const SIGNAGE_BANNER_ADDED = 'SIGNAGE_BANNER_ADDED';
+export const SIGNAGE_BANNER_DELETED = 'SIGNAGE_BANNER_DELETED';
+export const SIGNAGE_UPDATED = 'SIGNAGE_UPDATED';
+export const SIGNAGE_STATIC_BANNER_UPDATED = 'SIGNAGE_STATIC_BANNER_UPDATED';
+
+const getAblyChannel = (summitId, locationId) => `SIGNAGE:${summitId}:${locationId}`;
+
+const realtimeAbly = new Ably.Realtime(process.env['ABLY_API_KEY']);
```

Comment:
if we dont configure ABLY_API_KEY the summit admin errors out
could we make it optional ?
like if we dont configure it just dont break the site ?

Reply:
also could we rename ABLY_API_KEY to SIGNAGE_ ABLY_API_KEY ?